include/tdi/common/tdi_init.hpp

```diff
@@ -100,12 +100,36 @@ class Device {
   std::map<std::string, std::unique_ptr<const TdiInfo>> tdi_info_map_;
 };
 
+class WarmInitOptions {
+ public:
+  virtual ~WarmInitOptions() = default;
+};
+
+class WarmInitImpl {
```

Comment:
Does it make sense to make the default ctor as protected? We can add a comment saying that Device drivers are required to derive and implement their version

Reply:
yeah, Good point.